server/app/auth/oidc/CustomOidcLogoutRequest.java

```diff
@@ -0,0 +1,87 @@
+package auth.oidc;
+
+import com.google.common.base.Strings;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import java.net.URI;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CustomOidcLogoutRequest extends LogoutRequest {
+  /** The optional post-logout redirection param and URI. */
+  private final String postLogoutRedirectParam;
+
+  private final URI postLogoutRedirectURI;
+
+  private final Map<String, String> extraParams;
+
+  /**
+   * Create new OIDC logout request with a optional redirect url, optional client id, and other
+   * params.
+   */
+  public CustomOidcLogoutRequest(
+      final URI uri,
+      final String postLogoutRedirectParam,
+      final URI postLogoutRedirectURI,
+      final Map<String, String> extraParams) {
+
+    super(uri, /* idTokenHint */ null, postLogoutRedirectURI, /* state */ null);
+
+    if (Strings.isNullOrEmpty(postLogoutRedirectParam)) {
+      // default to OIDC spec
```

Comment:
Not blocking: Consider linking to the entry in the spec to make it easy for one to quickly find this information again.